### PR TITLE
Line Layer

### PR DESCRIPTION
**Pull request by stefankarschti**
#1118

### CODE REVIEW

**Review by sjg-wdw — APPROVED** [review]
Looks like a good start!

File: include/mbgl/shaders/gl/circle.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Generated code, do not modify this file!
-// Generated on 2023-05-12T17:51:34.322Z by mwilsnd using shaders/generate_shader_code.js
```
Comment (alexcristici) [inline include/mbgl/shaders/gl/circle.hpp:2]: In order to avoid conflicts it will be good to revert these files, that doesn't have any important changes.

Comment (TimSylvester) [reply include/mbgl/shaders/gl/circle.hpp:2]: I've been avoiding committing those, but it's a bit tedious.  The same basic info is in git-blame ("authors" pane in Xcode), so I would advocate removing the date and username.

Comment (stefankarschti) [reply include/mbgl/shaders/gl/circle.hpp:2]: +1

**Review by TimSylvester — APPROVED** [review]

File: platform/ios/platform/ios/app/MBXViewController.m [hunk 2]
```diff
@@ -1457,6 +1462,59 @@ -(void)toggleStyleLabelsLanguage
     [self.mapView.style localizeLabelsIntoLocale:_localizingLabels ? [NSLocale localeWithLocaleIdentifier:@"mul"] : nil];
 }
 
+- (void)styleLineGradient
```
Comment (stefankarschti) [inline platform/ios/platform/ios/app/MBXViewController.m:1465]: @louwers : this is added to test the line gradient

File: platform/ios/platform/ios/app/MBXViewController.m [hunk 3]
```diff
@@ -1909,6 +1967,9 @@ - (void)setStyles
     [self.styleNames addObject:@"MapLibre Basic"];
     [self.styleURLs addObject:[NSURL URLWithString:@"https://demotiles.maplibre.org/style.json"]];
 
+    [self.styleNames addObject:@"Line Pattern"];
+    [self.styleURLs addObject:[NSURL URLWithString:@"https://api.maptiler.com/maps/46b24e19-6503-4b2a-81a4-ea6d8431fc92/style.json?key=pdr3bbcne6eRBgLXQ1kg"]];
+
```
Comment (alexcristici) [inline platform/ios/platform/ios/app/MBXViewController.m:1972]: Probably this should be removed.

**Review by TimSylvester — APPROVED** [review]

File: src/mbgl/renderer/layers/render_line_layer.hpp [hunk 4]
```diff
@@ -8,18 +8,37 @@
...
+#include <memory>
+
 namespace mbgl {
 
 class LineProgram;
 class LineGradientProgram;
 class LineSDFProgram;
 class LinePatternProgram;
 
+namespace gfx {
```
Comment (mwilsnd) [inline src/mbgl/renderer/layers/render_line_layer.hpp:21]: Wrap these in `#if MLN_DRAWABLE_RENDERER`, keeping forward decls out of the legacy path is helpful for catching other places we may forget to exclude code.

**Review by mwilsnd — CHANGES_REQUESTED** [review]

**Review by mwilsnd — APPROVED** [review]

**Pull request merged**